rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/nodesource/NodeSource.java

```diff
@@ -114,9 +114,13 @@
     /** unique name of the source */
     private final String name;
 
-    private final InfrastructureManager infrastructureManager;
+    private InfrastructureManager infrastructureManager;
 
-    private final NodeSourcePolicy nodeSourcePolicy;
+    // The policy is not final: when the node source is defined, the policy is
+    // not an activity. When the node source is deployed, the policy is turned
+    // active, so we need to update the nodeSourcePolicy reference with the
+    // policy stub
+    private NodeSourcePolicy nodeSourcePolicy;
```

Comment:
alternatively you can add a separate field activeNodeSourcePolicy, null originally. This would make it easier later when you want to undeploy a node source

Reply:
ok, I applied this